### PR TITLE
Add telemetry using the extension API

### PR DESCRIPTION
**Pull request by gbraad**
This PR adds telemetry to the extension, utilizing the functionality provided by Podman Desktop.

### CODE REVIEW

**Comment by gbraad** [comment]
It finds formatting issues, but running the same command gives me:

```
Checking formatting...
[warn] src\command.ts
[warn] src\constants.ts
[warn] src\crc-cli.ts
[warn] src\crc-console.ts
[warn] src\crc-delete.ts
[warn] src\crc-setup.ts
[warn] src\crc-start.ts
[warn] src\crc-status.ts
[warn] src\crc-stop.ts
[warn] src\daemon-commander.ts
[warn] src\detection-checks.ts
[warn] src\dev-terminal.ts
[warn] src\extension.ts
[warn] src\install\base-install.ts
[warn] src\install\crc-install.ts
[warn] src\install\mac-install.ts
[warn] src\install\win-install.ts
[warn] src\log-provider.ts
[warn] src\logger.ts
[warn] src\login-commands.ts
[warn] src\preferences.ts
[warn] src\util.ts
[warn] Code style issues found in 22 files. Forgot to run Prettier?
```

instead of the 3 found.

File: src/extension.ts [hunk 1]
```diff
@@ -76,13 +77,14 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
...
 
   const providerLifecycle: extensionApi.ProviderLifecycle = {
-    status: () => crcStatus.getProviderStatus(),
-
+    status: () => {
+      return crcStatus.getProviderStatus();
+    },
     start: context => {
-      return startCrc(context.log);
+      return startCrc(context.log, this.telemetryLogger);
```
Comment (evidolob) [inline src/extension.ts:84]: I think you shouldn’t use `this.` to pass `telemetryLogger` in to `startCrc` function, as 
`telemetryLogger` defined as local variable, not like field of an object, so you passing `undefined`.

File: src/extension.ts [hunk 2]
```diff
@@ -192,10 +195,10 @@ async function registerOpenShiftLocalCluster(
         return deleteCrc();
       },
       start: ctx => {
-        return startCrc(ctx.log);
+        return startCrc(ctx.log, this.telemetryLogger);
```
Comment (evidolob) [inline src/extension.ts:198]: Same there

File: src/extension.ts [hunk 3]
```diff
@@ -76,13 +77,14 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
...
+    status: () => {
+      return crcStatus.getProviderStatus();
+    },
     start: context => {
-      return startCrc(context.log);
+      return startCrc(context.log, this.telemetryLogger);
     },
     stop: () => {
-      return stopCrc();
+      return stopCrc(this.telemetryLogger);
```
Comment (evidolob) [inline src/extension.ts:87]: Same there, just avoid using `this.`

File: src/extension.ts [hunk 4]
```diff
@@ -192,10 +195,10 @@ async function registerOpenShiftLocalCluster(
         return deleteCrc();
       },
       start: ctx => {
-        return startCrc(ctx.log);
+        return startCrc(ctx.log, this.telemetryLogger);
       },
       stop: () => {
-        return stopCrc();
+        return stopCrc(this.telemetryLogger);
```
Comment (evidolob) [inline src/extension.ts:201]: and there...

**Comment by gbraad** [comment]
Running `prettier` on a Linux machine, as on Windows it wants to modify all line-endings

**Comment by gbraad** [comment]
@evidolob PTAL.

> Error: The action has timed out.

happens quite often

**Review by evidolob — APPROVED** [review]

**Comment by gbraad** [comment]
macOS build will most likely time out again.

**Pull request merged**